### PR TITLE
Add References Values

### LINKED ISSUE #170: Get `ReferenceValues`
The Project Team provides reference values for the verifier to verify the values before endorsing a binary.

The Project Team gives the values in a .toml file matching the [`ProvenanceIR`](https://github.com/project-oak/transparent-release/blob/a7f89ea35e934e6d3f929c64770d94896ba8f1d4/internal/verifier/verifier.go#L184).

We want to:
- add functionality to parse the reference values,
- add a example reference in `testdata` e.g., for [provenance.json](https://github.com/project-oak/transparent-release/blob/a7f89ea35e934e6d3f929c64770d94896ba8f1d4/testdata/provenance.json).

We could 
- add functionality to use reference values in the endorser.

### PR DESCRIPTION
**Pull request by mariaschett**
Fixes #170  and partly #166.

### CODE REVIEW

File: testdata/reference_values.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+binary_256sha_digests = ["322527c0260e25f0e9a2595bd0d71a52294fe2397a7af76165190fd98de8920d"]
+want_build_cmds = true
+builder_image_256sha_digests = ["9e2ba52487d945504d250de186cb4fe2e3ba023ed2921dd6ac8b97ed43e76af9", "y"]
```
Comment (rbehjati) [inline testdata/reference_values.toml:3]:
```suggestion
builder_image_sha256_digests = ["9e2ba52487d945504d250de186cb4fe2e3ba023ed2921dd6ac8b97ed43e76af9", "y"]
```

Comment (rbehjati) [reply testdata/reference_values.toml:3]: We could also think of the the following format. Let's think and discuss this separately:

```
builder_image_igests = ["sha256:9e2ba52487d945504d250de186cb4fe2e3ba023ed2921dd6ac8b97ed43e76af9", "y"]
```

**Review by rbehjati — APPROVED** [review]
LGTM. Thanks :)

My main comment is on the field names in the toml file.

File: internal/common/common.go [hunk 2]
```diff
@@ -63,6 +63,16 @@ type RepoCheckoutInfo struct {
 	Logs string
 }
 
+// ReferenceValues given by the product team to verify provenances against.
+type ReferenceValues struct {
+	// The digests of the binaries whose provenance the product team wants to verify.
+	BinarySHA256Digests []string `toml:"binary_256sha_digests"`
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply internal/common/common.go]: Done, thank you!

File: internal/common/common.go [hunk 3]
```diff
@@ -63,6 +63,16 @@ type RepoCheckoutInfo struct {
...
+	// If true the product teams wants the provenance to have a non-empty build command.
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply internal/common/common.go]: Done! Thank you!

File: internal/common/common.go [hunk 4]
```diff
@@ -63,6 +63,16 @@ type RepoCheckoutInfo struct {
...
+	WantBuildCmds bool `toml:"want_build_cmds"`
+	// The digests of the builder images the product team trusts to build the binary.
+	BuilderImageSHA256Digests []string `toml:"builder_image_256sha_digests"`
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply internal/common/common.go]: Done, I don't know why I swapped those!

File: internal/endorser/endorser.go [hunk 5]
```diff
@@ -52,7 +53,7 @@ func GenerateEndorsement(referenceValues verifier.ProvenanceIR, validityDuration
 // (2) Any of the provenances cannot be loaded (e.g., invalid URI),
 // (3) Any of the provenances is invalid (see verifyProvenances for details on validity),
 // (4) Provenances do not match (e.g., have different binary names).
-func loadAndVerifyProvenances(provenanceURIs []string, referenceValues verifier.ProvenanceIR) (*amber.VerifiedProvenanceSet, error) {
+func loadAndVerifyProvenances(provenanceURIs []string, referenceValues common.ReferenceValues) (*amber.VerifiedProvenanceSet, error) {
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply internal/endorser/endorser.go]: Agree, swapped, also in `verifyProvenances`.

File: testdata/reference_values.toml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+binary_256sha_digests = ["322527c0260e25f0e9a2595bd0d71a52294fe2397a7af76165190fd98de8920d"]
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply testdata/reference_values.toml]: No!

File: testdata/reference_values.toml (hunk 1 above)
Comment (mariaschett) [reply testdata/reference_values.toml]: Sounds like a good idea, if we have other digests! Yes, let's discuss this seperately, but leave it for now.

**Review by mariaschett — COMMENTED** [review]
Thank you for the quick review!

**Pull request merged**